src/main/java/ee/tuleva/onboarding/payment/PaymentController.java

```diff
@@ -0,0 +1,66 @@
+package ee.tuleva.onboarding.payment;
+
+import ee.tuleva.onboarding.auth.principal.AuthenticatedPerson;
+import ee.tuleva.onboarding.currency.Currency;
+import ee.tuleva.onboarding.epis.EpisService;
+import ee.tuleva.onboarding.epis.contact.ContactDetails;
+import io.swagger.v3.oas.annotations.Operation;
+import java.math.BigDecimal;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.core.annotation.AuthenticationPrincipal;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+
+@Controller
+@RequestMapping("/v1/payments")
+@Slf4j
+@RequiredArgsConstructor
+public class PaymentController {
+
+  @Value("${frontend.url}")
+  private String frontendUrl;
+
+  private final PaymentProviderService paymentProviderService;
+  private final EpisService episService;
+  private final PaymentInternalReferenceService paymentInternalReferenceService;
+
+  @GetMapping("/link")
+  @Operation(summary = "Create a payment")
+  public String createPayment(
+      @AuthenticationPrincipal AuthenticatedPerson authenticatedPerson,
+      @RequestParam Currency currency,
+      @RequestParam BigDecimal amount,
+      @RequestParam Bank bank) {
+
+    ContactDetails contactDetails = episService.getContactDetails(authenticatedPerson);
```

Comment:
I would extract this business logic into the Service layer i.e. `PaymentService`. A Controller should have no business logic, it just exposes the Service over the web, using the HTTP transport mechanism.